### PR TITLE
wtip-003: implement airlines PoC

### PR DESCRIPTION
**Pull request by ondrejchmelar**
https://github.com/windingtree/wtips/blob/master/WTips/wtip-003.md
https://github.com/windingtree/wtips/issues/10

### CODE REVIEW

File: docs/swagger.yaml [hunk 1]
```diff
@@ -164,49 +170,15 @@ components:
...
-                  amount:
-                    description: Cost of cancellation in percents of the final price, 100 means non refundable. This means how much money will the hotel keep.
-                    type: number
-                    format: float
+          $ref: '#/components/schemas/Pricing'
         # billing: TBD (cash on arrival, card on arrival, online payment, crypto)
         # guarantee: TBD (can be for example credit card info)
     Customer:
-      description: Person responsible for this booking who should be contacted by the hotel if needed
+      description: Person responsible for this booking who should be contacted by the hotel/airline supplier if needed
```
Comment (JirkaChadima) [inline docs/swagger.yaml]: Maybe we can simplify this just to supplier?

File: docs/swagger.yaml [hunk 2]
```diff
@@ -247,7 +219,47 @@ components:
...
+              from:
+                description: After which date does this cancellation fee apply (inclusive).
+                type: string
+                format: date
+              to:
+                description: Until which date does this cancellation policy apply (inclusive).
+                type: string
+                format: date
+              amount:
+                description: Cost of cancellation in percents of the final price, 100 means non refundable. This means how much money will the hotel keep.
```
Comment (JirkaChadima) [inline docs/swagger.yaml]: Replace hotel with supplier

File: src/config/envvar.js [hunk 3]
```diff
@@ -45,9 +45,9 @@ module.exports = {
   updateAvailability: convertEnvVarToBoolean(process.env.UPDATE_AVAILABILITY, true),
```
Comment (JirkaChadima) [inline src/config/envvar.js]: Does it make sense to keep the same name for airlines? I think there's a different concept than availability there, isn't it?

File: src/services/adapters/airline-adapter.js [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+    }
+
+    if (total < (price - EPSILON)) {
+      throw new adapter.InvalidPriceError(`The total is too low, expected ${price}`);
+    }
+  }
+
+  /**
+   * Check admissibility wrt. the suggested price,
+   * room availability and cancellation fees.
```
Comment (JirkaChadima) [inline src/services/adapters/airline-adapter.js]: `room availability` does not make sense wrt airlines

File: src/config/envvar.js (hunk 3 above)
Comment (ondrejchmelar) [reply src/config/envvar.js]: It is a bit different (available rooms vs available seats on a flight) but still related to availability, so I'd say the name works for both, doesn't it?

**Comment by ondrejchmelar** [comment]
IIRC `finalizationUrl` is supposed to be used for payment, @czervenka is that right? As we're not dealing with payment for now I think it is ok it is not used.

File: src/config/envvar.js (hunk 3 above)
Comment (JirkaChadima) [reply src/config/envvar.js]: Ok, cool. Let's leave it at that.

**Comment by czervenka** [comment]
> IIRC `finalizationUrl` is supposed to be used for payment, @czervenka is that right? As we're not dealing with payment for now I think it is ok it is not used.

Yes, it is meant for payment (or any other redirection after successful booking). We don't need to implement it on booking api server implementation, but IMO it is good to keep it in the booking protocol.

**Comment by JirkaChadima** [comment]
OK, let's keep it in the specs. Thanks!

**Comment by JirkaChadima** [comment]
I think this is waiting mainly for https://github.com/windingtree/wiki/pull/83

**Pull request merged**